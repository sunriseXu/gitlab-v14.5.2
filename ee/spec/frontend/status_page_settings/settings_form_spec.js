import { shallowMount } from '@vue/test-utils';
import StatusPageSettingsForm from 'ee/status_page_settings/components/settings_form.vue';
import createStore from 'ee/status_page_settings/store';

describe('Status Page settings form', () => {
  let wrapper;
  const store = createStore();

  const findForm = () => wrapper.find({ ref: 'settingsForm' });
  const findToggleButton = () => wrapper.find({ ref: 'toggleBtn' });
  const findSectionHeader = () => wrapper.find({ ref: 'sectionHeader' });
  const findSectionSubHeader = () => wrapper.find({ ref: 'sectionSubHeader' });

  beforeEach(() => {
    wrapper = shallowMount(StatusPageSettingsForm, { store });
  });

  afterEach(() => {
    if (wrapper) {
      wrapper.destroy();
    }
  });

  describe('default state', () => {
    it('should match the default snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('renders header text', () => {
    expect(findSectionHeader().text()).toBe('Status page');
  });

  describe('expand/collapse button', () => {
    it('renders as an expand button by default', () => {
      expect(findToggleButton().text()).toBe('Expand');
    });
  });

  describe('sub-header', () => {
    it('renders descriptive text', () => {
      expect(findSectionSubHeader().text()).toContain(
        'Configure file storage settings to link issues in this project to an external status page.',
      );
    });
  });

  describe('form', () => {
    beforeEach(() => {
      jest.spyOn(wrapper.vm, 'updateStatusPageSettings').mockImplementation();
    });

    describe('submit button', () => {
      it('submits form on click', () => {
        findForm().trigger('submit');
        expect(wrapper.vm.updateStatusPageSettings).toHaveBeenCalled();
      });
    });
  });
});
