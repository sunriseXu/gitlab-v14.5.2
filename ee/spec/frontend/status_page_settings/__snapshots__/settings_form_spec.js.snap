// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Page settings form default state should match the default snapshot 1`] = `
<section
  class="settings no-animate js-status-page-settings"
  id="status-page"
>
  <div
    class="settings-header"
  >
    <h4
      class="settings-title js-settings-toggle js-settings-toggle-trigger-only"
    >
      
      Status page
    
    </h4>
     
    <gl-button-stub
      buttontextclasses=""
      category="primary"
      class="js-settings-toggle"
      icon=""
      size="medium"
      variant="default"
    >
      Expand
    </gl-button-stub>
     
    <p>
      
      Configure file storage settings to link issues in this project to an external status page.
    
    </p>
  </div>
   
  <div
    class="settings-content"
  >
    <p>
      <gl-sprintf-stub
        message="To publish incidents to an external status page, GitLab stores a JSON file in your Amazon S3 account at a location that your external status page service can access. Make sure to also set up %{docsLink}"
      />
    </p>
     
    <form>
      <gl-form-group-stub
        class="gl-pl-0 mb-3"
        labeldescription=""
      >
        <gl-form-checkbox-stub>
          <span
            class="bold"
          >
            Active
          </span>
        </gl-form-checkbox-stub>
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-8 col-md-9 gl-pl-0 mb-3"
        label="Status page URL"
        label-for="status-page-url"
        label-size="sm"
        labeldescription=""
      >
        <gl-form-input-stub
          id="status-page-url"
          value=""
        />
         
        <p
          class="form-text text-muted"
        >
          <gl-link-stub
            href="/help/operations/incident_management/status_page#configure-gitlab-with-cloud-provider-information"
            target="_blank"
          >
            
            Configuration help
          
          </gl-link-stub>
        </p>
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-8 col-md-9 gl-pl-0 mb-3"
        label="S3 Bucket name"
        label-for="status-page-s3-bucket-name"
        label-size="sm"
        labeldescription=""
      >
        <gl-form-input-stub
          id="status-page-s3-bucket-name"
          value=""
        />
         
        <p
          class="form-text text-muted"
        >
          <gl-sprintf-stub
            message="Bucket %{docsLink}"
          />
        </p>
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-8 col-md-9 gl-pl-0 mb-3"
        label="AWS region"
        label-for="status-page-aws-region"
        label-size="sm"
        labeldescription=""
      >
        <gl-form-input-stub
          id="status-page-aws-region"
          placeholder="example: us-west-2"
          value=""
        />
         
        <p
          class="form-text text-muted"
        >
          <gl-sprintf-stub
            message="AWS %{docsLink}"
          />
        </p>
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-8 col-md-9 gl-pl-0 mb-3"
        label="AWS access key ID"
        label-for="status-page-aws-access-key"
        label-size="sm"
        labeldescription=""
      >
        <gl-form-input-stub
          id="status-page-aws-access-key "
          value=""
        />
      </gl-form-group-stub>
       
      <gl-form-group-stub
        class="col-8 col-md-9 gl-pl-0 mb-3"
        label="AWS Secret access key"
        label-for="status-page-aws-secret-access-key"
        label-size="sm"
        labeldescription=""
      >
        <gl-form-input-stub
          id="status-page-aws-secret-access-key "
          value=""
        />
      </gl-form-group-stub>
       
      <gl-button-stub
        buttontextclasses=""
        category="primary"
        class="js-no-auto-disable"
        icon=""
        size="medium"
        type="submit"
        variant="confirm"
      >
        
        Save changes
      
      </gl-button-stub>
    </form>
  </div>
</section>
`;
